library/std/build.rs

```diff
@@ -48,7 +53,7 @@ fn main() {
         // - mipsel-sony-psp
         // - nvptx64-nvidia-cuda
         // - arch=avr
-        // - JSON targets
+        // - JSON targets not describing an excluded target above.
```

Comment:
As I understand it restricted-std is meant to prevent being able to use libstd on stable for targets that we don't actually support libstd support for once `-Zbuild-std` is stabilized. When using custom json targets, IMO it shouldn't be stable as libstd changes may break support in the future even if it happens to work currently.

Reply:
Hm, I think that seems reasonable. We have a long-standing issue about making custom target specs entirely unstable (https://github.com/rust-lang/rust/issues/71009), which seems like the better fix to me. Or an explicit guard here if we want (not sure how feasible that is). I'd be ok cutting this specific change.

Reply:
Giving `--target=file.json some unstable-machinery at the CLI level, to me, would clearly imply that std doesn't support JSON targets in the same way Rust doesn't.

At the moment I'm not sure how a build script would detect a JSON target without the old broken TARGET behaviour